test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,615 @@
+		await page.keyboard.press( 'Escape' );
+
+		// Navigate to the block.
+		await page.keyboard.press( 'Tab' );
+
+		// Copy the colored text.
+		await pageUtils.pressKeyWithModifier( 'primary', 'c' );
+
+		// Collapse the selection to the end.
+		await page.keyboard.press( 'ArrowRight' );
+
+		// Create a new paragraph.
+		await page.keyboard.press( 'Enter' );
+
+		// Paste the colored text.
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should paste paragraph contents into list', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+
+		// Create two lines of text in a paragraph.
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
+		await page.keyboard.type( '2' );
+
+		// Select all and copy.
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'c' );
+
+		// Collapse the selection to the end.
+		await page.keyboard.press( 'ArrowRight' );
+
+		// Create a list.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '* ' );
+
+		// Paste paragraph contents.
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should paste list contents into paragraph', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+
+		// Create an indented list of two lines.
+		await page.keyboard.type( '* 1' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( ' 2' );
+
+		// Select all and copy.
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'c' );
+
+		// Collapse the selection to the end.
+		await page.keyboard.press( 'ArrowRight' );
+
+		// Create a paragraph.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+
+		// Paste paragraph contents.
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should navigate arround emoji', async ( { editor, page } ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '🍓' );
+		// Only one press on arrow left should be required to move in front of
+		// the emoji.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.type( '1' );
+
+		// Expect '1🍓'.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should run input rules after composition end', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		// Puppeteer doesn't support composition, so emulate it by inserting
+		// text in the DOM directly, setting selection in the right place, and
+		// firing `compositionend`.
+		// See https://github.com/puppeteer/puppeteer/issues/4981.
```

Comment:
```suggestion
		// See https://github.com/microsoft/playwright/issues/5777.
```

That's the relevant Playwright issue.